src/client/sandbox/fetch.ts

```diff
@@ -244,8 +234,44 @@ export default class FetchSandbox extends SandboxBase {
             return nativeMethods.headersForEach.apply(this, args);
         });
 
-        overrideFunction(window.Headers.prototype, 'get', FetchSandbox._createAccessorWrapper(nativeMethods.headersGet));
-        overrideFunction(window.Headers.prototype, 'set', FetchSandbox._createAccessorWrapper(nativeMethods.headersSet));
-        overrideFunction(window.Headers.prototype, 'has', FetchSandbox._createAccessorWrapper(nativeMethods.headersHas));
+        overrideFunction(window.Headers.prototype, 'get', function (...args) {
+            const [headerName] = args;
+
+            args[0] = transformHeaderNameToInternal(headerName);
+
+            const result = nativeMethods.headersGet.apply(this, args);
+
+            if (result === null) {
+                args[0] = headerName;
+
+                return nativeMethods.headersGet.apply(this, args);
+            }
+
+            return result;
+        });
+
+        overrideFunction(window.Headers.prototype, 'has', function (...args) {
+            const [headerName] = args;
+
+            args[0] = transformHeaderNameToInternal(headerName);
+
+            const result = nativeMethods.headersHas.apply(this, args);
+
+            if (!result) {
+                args[0] = headerName;
+
+                return nativeMethods.headersHas.apply(this, args);
+            }
+
+            return result;
+        });
+
+        overrideFunction(window.Headers.prototype, 'has', function (...args) {
```

Comment:
Shouldn't it be `'set'`?

Reply:
Yes, it should be `set`. Also, add a test for the `set` method call.